### PR TITLE
[addons] move GetTempAddonBasePath() from binary to normal AddonManager

### PR DESCRIPTION
**Pull request by AlwinEsch**
## Description
This is the first step to remove the CBinaryAddonManager. Since removal
of cpluff this no more direct needed.

Therefore this request, alone, since this is currently used for the binaries, I also want to ask whether there is something similar in the ***Python*** addons, which gives them a temporary path?
If so, we could combine the two so that there is more order in the mess.

My goal is to make the addon interface cleaner and then try to make it more stable. 
Because there is so much mix between things at the moment, it becomes difficult to create a centrally controlled path that takes care of what (e.g. ongoing addon changes while Kodi is running...)

I may also change the name "GetTempAddonBasePath" to something else. Can you think of something or do you think it's OK?

## Types of change

- [ ] **Bug fix** (non-breaking change which fixes an issue)
- [ ] **Clean up** (non-breaking change which removes non-working, unmaintained functionality)
- [ ] **Improvement** (non-breaking change which improves existing functionality)
- [ ] **New feature** (non-breaking change which adds functionality)
- [ ] **Breaking change** (fix or feature that will cause existing functionality to change)
- [ ] **Cosmetic change** (non-breaking change that doesn't touch code)
- [ ] **None of the above** (please explain below)

## Checklist:

- [ ] My code follows the **[Code Guidelines](https://github.com/xbmc/xbmc/blob/master/docs/CODE_GUIDELINES.md)** of this project 
- [ ] My change requires a change to the documentation, either Doxygen or wiki
- [ ] I have updated the documentation accordingly
- [ ] I have read the **[Contributing](https://github.com/xbmc/xbmc/blob/master/docs/CONTRIBUTING.md)** document
- [ ] I have added tests to cover my change
- [ ] All new and existing tests passed

### CODE REVIEW

File: xbmc/addons/AddonManager.cpp [hunk 1]
```diff
@@ -131,6 +131,10 @@ bool CAddonMgr::Init()
 void CAddonMgr::DeInit()
 {
   m_database.Close();
+
+  /* If temporary directory was used from addon delete them */
```
Comment (garbear) [inline xbmc/addons/AddonManager.cpp]:
```c++
/* If temporary directory was used from add-on, delete it */
```

File: xbmc/addons/AddonManager.h [hunk 2]
```diff
@@ -238,6 +238,16 @@ namespace ADDON
     bool GetAddonInfos(AddonInfos& addonInfos, TYPE type);
     const AddonInfoPtr GetAddonInfo(const std::string& id, TYPE type = ADDON_UNKNOWN);
 
+    /*!
+     * @brief To get the path where temporary addon parts can be becomes stored
```
Comment (garbear) [inline xbmc/addons/AddonManager.h]:
```c++
* @brief Get the path where temporary add-on files are stored
```

File: xbmc/addons/AddonManager.h [hunk 3]
```diff
@@ -238,6 +238,16 @@ namespace ADDON
...
+     *
+     * @return the base path used for temporary addon paths
+     *
+     * @warning the folder and his contents becomes deleted during stop and
```
Comment (garbear) [inline xbmc/addons/AddonManager.h]:
```
* @warning the folder and its contents are deleted when Kodi is closed
```

File: xbmc/addons/AddonManager.h [hunk 4]
```diff
@@ -258,6 +268,10 @@ namespace ADDON
     std::set<std::string> m_systemAddons;
     std::set<std::string> m_optionalAddons;
     ADDON_INFO_LIST m_installedAddons;
+
+    // Temporary path where can given to addons, his content becomes during
```
Comment (garbear) [inline xbmc/addons/AddonManager.h]:
```
// Temporary path given to add-ons, whose content is deleted when Kodi is stopped
```

**Review by garbear — APPROVED** [review]
Left some grammar comments, and the rest looks good

File: xbmc/addons/AddonManager.h (hunk 4 above)
Comment (AlwinEsch) [reply xbmc/addons/AddonManager.h]: Thanks a lot, my bad english is corrected :smile: !

**Comment by ronie** [comment]
> I also want to ask whether there is something similar in the _**Python**_ addons, which gives them a temporary path?

nope, there's no such thing as far as i know.

**Comment by AlwinEsch** [comment]
Thanks to you @ronie, would it make sense to have something there too?

**Comment by AlwinEsch** [comment]
e.g. to have:
```py
tmpPath = self.Addon.getTempAddonPath()
```
like this http://alwinesch.github.io/group__cpp__kodi.html#ga437ac949d8c4dbe95e9cb04830dd6ff8

**Comment by ronie** [comment]
> Thanks to you @ronie, would it make sense to have something there too?

we always tell python addon devs to store all their (temp) data in ../userdata/addon_data/addon.id
afaik there's no need for anything else..

**Comment by AlwinEsch** [comment]
jenkins build this please

**Pull request merged**